include/xtensor/xadapt.hpp

```diff
@@ -89,7 +89,7 @@ namespace xt
      */
     template <class C, std::size_t N, layout_type L = DEFAULT_LAYOUT>
     xtensor_adaptor<C, N, L>
-    xadapt(C& container, const std::array<typename C::size_type, N>& shape, layout_type l = L);
+    xadapt(C& container, const stat_shape<typename C::size_type, N>& shape, layout_type l = L);
```

Comment:
Argh, I can't comment on the lines above, but there is still a `std::enable_if_t<!detail::is_array<SC>::value`... which obviously should be replaced. Maybe we can have something like `detail::is_static_container` or `detail::is_static_size`? 
This currently prevents building of the xadapt tests with GCC.